src/main/java/bob/Ui.java

```diff
@@ -0,0 +1,208 @@
+package bob;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+/**
+ * Represents a parser that deals with interactions with the user
+ */
+public class Ui {
+
+    private static Parser parser;
+
+    public Ui() {
+        this.parser = new Parser();
+    }
+
+    /**
+     * Marks a task as completed according to specified task index
+     *
+     * @param markNo index of the task in the list to be marked
+     */
+    public static void markTask(TaskList list, int markNo) {
+        if (markNo > 0 && markNo <= list.size()) {
+            System.out.println("Nice! I've marked this task as done:");
+            list.get(markNo - 1).markAsDone();
+            System.out.println(list.get(markNo - 1).toString());
+        } else {
+            System.out.println("Sorry, there is no such task!");
+        }
+    }
+
+    /**
+     * Deletes a task from the list according to specified task index
+     *
+     * @param deleteNo index of the task in the list to be deleted
+     */
+    public static void deleteTask(TaskList list, int deleteNo) {
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(list.get(deleteNo - 1).toString());
+        list.remove(deleteNo - 1);
+        System.out.println("Now you have " + String.valueOf(list.size()) + " tasks in the list.");
+    }
+
+    /**
+     * Finds task according to the specified keyword and lists them out
+     * 
+     * @param list the list containing all tasks
+     * @param keyword the keyword used to filter tasks
+     */
+    public static void findTask(TaskList list, String keyword) {
+        ArrayList<Task> matches = new ArrayList<Task>();
+
+        for (int i = 0; i < list.size(); i++) {
+            if (list.get(i).toString().contains(keyword)) {
+                matches.add(list.get(i));
+            }
+        }
+
+        System.out.println("Here are the matching tasks in your list:");
+        for (int i = 1; i <= matches.size(); i++) {
+            System.out.println(i + ". " + list.get(i - 1).toString());
+        }
+    }
+
+    /**
+     * Adds a task into the list of tasks
+     *
+     * @param newTask the new Task object that is to be added into the list
+     */
+    public static void addTask(TaskList list, Task newTask) {
+        list.add(newTask);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(newTask.toString());
+        System.out.println("Now you have " + String.valueOf(list.size()) + " tasks in the list.");
+    }
+
+    /**
+     * Checks what type of task is given in the input, Identifies the task name and dates/times (if applicable),
+     * then instantiates the Task of the specified type and adds it to list by calling addTask method.
+     *
+     * @param task the input string given
+     * @throws BobException
+     */
+    public static void checkAndAddTask(TaskList list, String task) throws BobException {
```

Comment:
Should some parts in this method be extracted out? As per [code quality guidelines](https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/codeQuality.html), we should avoid long methods.